src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -1184,15 +1199,64 @@ impl_commit_tree (RpmOstreeTreeComposeContext *self,
         glnx_prefix_error (error, "Handling group db");
     }
 
+  /* See comment above */
+  const gboolean use_txn = (getenv ("RPMOSTREE_COMMIT_NO_TXN") == NULL &&
+                            !repo_is_on_netfs (self->repo));
+
+  if (use_txn)
+    {
+      if (!ostree_repo_prepare_transaction (self->repo, NULL, cancellable, error))
+        return FALSE;
+    }
+
+  g_autofree char *parent_revision = NULL;
+  if (self->ref)
+    {
+      if (!ostree_repo_resolve_rev (self->repo, self->ref, TRUE, &parent_revision, error))
+        return FALSE;
+    }
+
   /* The penultimate step, just basically `ostree commit` */
   g_autofree char *new_revision = NULL;
-  if (!rpmostree_commit (self->rootfs_dfd, self->repo, self->ref, opt_write_commitid_to,
-                         metadata, gpgkey, selinux, self->devino_cache,
-                         &new_revision,
-                         cancellable, error))
+  if (!rpmostree_compose_commit (self->rootfs_dfd, self->repo, parent_revision,
+                                 metadata, gpgkey, selinux, self->devino_cache,
+                                 &new_revision, cancellable, error))
     return FALSE;
 
-  g_print ("%s => %s\n", self->ref, new_revision);
+  /* --write-commitid-to overrides writing the ref */
+  if (self->ref && !opt_write_commitid_to)
+    {
+      if (use_txn)
+        ostree_repo_transaction_set_ref (self->repo, NULL, self->ref, new_revision);
+      else
+        {
+          if (!ostree_repo_set_ref_immediate (self->repo, NULL, self->ref, new_revision,
+                                              cancellable, error))
+            return FALSE;
+        }
+    }
+
+  if (use_txn)
+    {
+      OstreeRepoTransactionStats stats = { 0, };
+      if (!ostree_repo_commit_transaction (self->repo, &stats, cancellable, error))
+        return glnx_prefix_error (error, "Commit");
+
+      g_print ("Wrote commit: %s\n", new_revision);
+      g_print ("Metadata Total: %u\n", stats.metadata_objects_total);
+      g_print ("Metadata Written: %u\n", stats.metadata_objects_written);
+      g_print ("Content Total: %u\n", stats.content_objects_total);
+      g_print ("Content Written: %u\n", stats.content_objects_written);
+      g_print ("Content Bytes Written: %" G_GUINT64_FORMAT "\n", stats.content_bytes_written);
+    }
+
+  if (opt_write_commitid_to)
+    {
+      if (!g_file_set_contents (opt_write_commitid_to, new_revision, -1, error))
+        return glnx_prefix_error (error, "While writing to '%s'", opt_write_commitid_to);
```

Comment:
We used to unconditionally print the `ref => csum` line for both cases, though that wasn't entirely appropriate since we're not updating the ref in this case. Though let's still print `new_revision` here to help when debugging build system?

Reply:
I added the checksum print above the txn stats.

Reply:
Ahh sorry, I missed that! I rebased it and moved the checksum print outside of the `if (txn)` conditional, so we always print it.